lib/client/ipam_block_reader_writer.go

```diff
@@ -309,24 +316,36 @@ func randomBlockGenerator(pool cnet.IPNet) func() *cnet.IPNet {
 	ones, size := pool.Mask.Size()
 	prefixLen := size - ones
 	numBlocks := int(math.Exp2(float64(prefixLen))) / blockSize
-
-	// Generate a randomly ordered slice of block indexes.
-	source := rand.NewSource(time.Now().UnixNano())
-	randm := rand.New(source)
-	idxs := randm.Perm(numBlocks)
 	i := 0
+	if !indexInit {
```

Comment:
I don't think you need all this `indexInit` logic - just store off the random initial index, and set `i := initialIndex`.

Each time `randomBlockGenerator` is called a new initial index will be set.